### ISSUE TITLE
Inconsistency of handling DataFrame creation and assignment

### ISSUE THREAD

**Issue by bkamins** [JuliaData/DataFrames.jl#1388]
When `DataFrame` is created it converts all arguments to `Vector`. However, assignment allows other structures (like `Range`) as columns in `DataFrame`. I knew about it fore some time, but it constantly bothers me so here is an issue 😄 .
Is it an intended feature or should we fix it?

Here is an example of the difference:
```
julia> using DataFrames

julia> DataFrame(x=1:3)
3×1 DataFrames.DataFrame
│ Row │ x │
├─────┼───┤
│ 1   │ 1 │
│ 2   │ 2 │
│ 3   │ 3 │

julia> df1 = DataFrame(x=1:3)
3×1 DataFrames.DataFrame
│ Row │ x │
├─────┼───┤
│ 1   │ 1 │
│ 2   │ 2 │
│ 3   │ 3 │

julia> df2 = DataFrame()
0×0 DataFrames.DataFrame


julia> df2[:x] = 1:3
1:3

julia> df1[:x]
3-element Array{Int64,1}:
 1
 2
 3

julia> df2[:x]
1:3
```

**Comment by nalimilan** [comment]
Good catch. I'm not sure what's the best choice. I'm slightly inclined to avoid converting ranges to vectors, given that it's easy to call `collect`, but that there's no way around if we make the conversion automatically.

**Comment by nalimilan** [comment]
Yes, it was introduced on purpose when we moved away from `DataArray` by default. Good point about the breakage when removing or adding rows, I hadn't thought about that. So maybe better keep that behavior and change assignment to do the same. After all the gain in memory use should be quite limited compared with the size of the full set of columns.

**Comment by bkamins** [comment]
OK - will make a PR and update docs accordingly.

**Comment by bkamins** [comment]
PR in #1392

**Comment by bkamins** [comment]
fixed by #1392
